.github/workflows/test.yml

```diff
@@ -0,0 +1,57 @@
+---
+## Test with multiple ruby images and frameworks
+name: test
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*.*.*
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-image:
+          - 'ruby:2.7'
+          - 'ruby:2.6'
+          - 'ruby:2.5'
+          - 'ruby:2.4'
+          - 'ruby:2.3'
+          - 'jruby:9.2'
+        framework:
+          - 'rails-6.1'
+          - 'rails-6.0'
+          - 'rails-5.2'
+          - 'sinatra-2.0'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7.2'
+      - uses: elastic/apm-pipeline-library/.github/actions/docker-login@current
+        if: github.event.pull_request.head.repo.full_name == github.repository # Only docker-login if the PR is not from a fork.
```

Comment:
FYI, I added this condition because `secrets.*` are not available if the PR is from a fork.